text/0000-any-stack-buildpacks.md

```diff
@@ -0,0 +1,49 @@
+# Meta
+[meta]: #meta
+- Name: Stackless Buildpacks
+- Start Date: 2020-07-26
+- Author(s): @sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This change removes the restriction that buildpacks must specify at least one stack in `buildpack.toml`. When buildpacks specify no stacks, the lifecycle will assume that the buildpack is valid on all stacks.
+
+# Motivation
+[motivation]: #motivation
+
+Some buildpacks are simple and/or don't tightly depend on underlying stack functionality. These buildpacks may not need to specify that they work on many different Linux distros.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The `buildpack.toml` file currently requires a `[[stacks]]` object list at the top-level. When this object list is empty or not present, this proposal requires that platforms and lifecycles allow the buildpack to be:
+1. Runnable on any stack, regardless of stack ID or mixins.
+2. Includable in any buildpackage, and reference-able in any meta-buildpack's `buildpack.toml` file.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Buildpack authors many inadvertently leave `[[stacks]]` out (by mistake, or due to lack of knowledge of stacks). If they have stack-specific logic in their buildpacks (e.g., the buildpack is a Ruby script), it would fail at build-time. If this happens to many buildpacks, end-users would suffer more failed buildpacks. More generally, the `[[stacks]]` array may turn into something that's only used to specify mixins in practice.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- We could add a special field that indicates any-stack compatibility, so that buildpack authors need to explicitly opt-in to the new behavior. This would prevent buildpack authors who don't know about stacks from inadvertently declaring any-stack compatibility without adequate testing.
```

Comment:
This seems like a good compromise. WRT platforms, maybe we just claim a set of stack ids for all, linux, windows? I'm not married to any of those names.

Reply:
Jumping on this bandwagon - if it could harm the UX, I'd rather it be gated/you have to opt in, so that buildpack authors don't ruin users trust in the platform.

Reply:
Done